### PR TITLE
GH-38300: [Dev][Docs] Update dev/README.md for the current merge_arrow_pr.py

### LINKED ISSUE #38300: [Dev][Docs] dev/merge_arrow_pr.py doesn't require "arrow" remote
### Describe the enhancement requested

https://github.com/apache/arrow/tree/main/dev#how-to-merge-a-pull-request says that we need "arrow" remote but it's no longer needed. Because we migrated to GitHub API from local Git merge by #13184.

### Component(s)

Developer Tools, Documentation

### PR DESCRIPTION
**Pull request by kou**
### Rationale for this change

dev/README.md is outdated.

### What changes are included in this PR?

* Don't require `arrow` remote
* Remove ARROW Jira related documents

### Are these changes tested?

No.

### Are there any user-facing changes?

No.
* Closes: #38300

### CODE REVIEW

File: dev/README.md [hunk 1]
```diff
@@ -113,7 +95,7 @@ Would you like to update the associated JIRA? (y/n): y
 Enter comma-separated fix version(s) [11.0.0]:
 ```
 
-```
+```text
```
Comment (raulcd) [inline dev/README.md:98]: This block seems duplicated

Comment (raulcd) [reply dev/README.md:98]: I mean this block appears twice, right?
```
Author 1: Name
Pull request #X merged!
Merge hash: #hash
Would you like to update the associated JIRA? (y/n): y
Enter comma-separated fix version(s) [11.0.0]:
```

Comment (kou) [reply dev/README.md]: Ah, they aren't duplicated. One is for Jira and another is for GitHub issue.
We can remove Jjira's one. I've removed it.

**Comment by kou** [comment]
Can we merge this?

**Review by raulcd — APPROVED** [review]

**Pull request merged**